### PR TITLE
chore: replace `var` with `let` and `const`

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

**Comment by DominicKramer** [comment]
PTAL

File: test/test-servers/koa_scaffold_server.ts [hunk 1]
```diff
@@ -16,11 +16,12 @@
 // jscs doesn't understand koa..
 // jscs:disable
 
-var errorHandler = require('../../src/index.js')({
+import {ErrorReporting} from '../../src/index';
```
Comment (JustinBeckwith) [inline test/test-servers/koa_scaffold_server.ts]: technically don't need the `/index` at the end :)

File: test/test-servers/koa_scaffold_server.ts [hunk 2]
```diff
@@ -16,11 +16,12 @@
...
+const errorHandler = new ErrorReporting({
   onUncaughtException: 'report',
-});
-var koa = require('koa');
-var app = koa();
+} as {});
```
Comment (JustinBeckwith) [inline test/test-servers/koa_scaffold_server.ts]: What's with the `as {}`?

File: test/test-servers/koa_scaffold_server.ts [hunk 3]
```diff
@@ -31,18 +32,18 @@ app.use(function*(this, next) {
...
-  var ms = new Date() as any - start;
+  const ms = new Date() as any - start;
   this.set('X-Response-Time', ms + 'ms');
 });
 
 // logger
 
 app.use(function*(this, next) {
-  var start = new Date() as any;
+  const start = new Date() as any;
```
Comment (JustinBeckwith) [inline test/test-servers/koa_scaffold_server.ts]: This doesn't feel right.  Why are you casting the Date to any?

File: test/unit/configuration.ts [hunk 4]
```diff
@@ -16,20 +16,20 @@
 
 import * as assert from 'assert';
 import * as is from 'is';
-var isNumber = is.number;
+const isNumber = is.number;
```
Comment (JustinBeckwith) [inline test/unit/configuration.ts]: There's an @types package for `is`, could you install and use that?

File: test/unit/interfaces/hapi.ts [hunk 5]
```diff
@@ -16,8 +16,8 @@
 
 import has = require('lodash.has');
 import * as is from 'is';
-var isFunction = (is as {} as {fn: Function}).fn;
-var isObject = is.object;
+const isFunction = (is as {} as {fn: Function}).fn;
```
Comment (JustinBeckwith) [inline test/unit/interfaces/hapi.ts]: lets use the @types

File: test/unit/interfaces/manual.ts [hunk 6]
```diff
@@ -17,28 +17,28 @@
...
+const config = new Configuration({});
 (config as {} as {lacksCredentials: Function}).lacksCredentials = function() {
   return false;
 };
 import {ErrorMessage} from '../../../src/classes/error-message';
 import { RequestHandler } from '../../../src/google-apis/auth-client';
 import { Logger } from '@google-cloud/common';
 import { RequestInformationContainer } from '../../../src/classes/request-information-container';
-// var nock = require('nock');
+// const nock = require('nock');
```
Comment (JustinBeckwith) [inline test/unit/interfaces/manual.ts]: remove?

Comment (DominicKramer) [reply test/unit/interfaces/manual.ts]: Done

File: test/test-servers/koa_scaffold_server.ts (hunk 1 above)
Comment (DominicKramer) [reply test/test-servers/koa_scaffold_server.ts]: Done

File: test/test-servers/koa_scaffold_server.ts (hunk 2 above)
Comment (DominicKramer) [reply test/test-servers/koa_scaffold_server.ts]: This is needed because the constructor is using an invalid configuration option.  I think it would be better to address this in a follow-up PR.

File: test/test-servers/koa_scaffold_server.ts (hunk 3 above)
Comment (DominicKramer) [reply test/test-servers/koa_scaffold_server.ts]: Yes, we should be using `Date.now()`.  I've added a TODO since I think this should be addressed in a follow-up PR.

File: test/unit/configuration.ts (hunk 4 above)
Comment (DominicKramer) [reply test/unit/configuration.ts]: I think this should be using `@types/is`.  Otherwise I would need an explicit type for `isNumber`.  I left the variables as they are since I didn't want to do structural changes during the TypeScript translation.

**Comment by DominicKramer** [comment]
PTAL

File: test/unit/interfaces/hapi.ts (hunk 5 above)
Comment (JustinBeckwith) [reply test/unit/interfaces/hapi.ts]: FYI - I just added it:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24542

File: test/test-servers/express_scaffold_server.ts [hunk 7]
```diff
@@ -15,26 +15,26 @@
...
 // eslint-disable-next-line no-console
-var log = console.log;
+const log = console.log;
 
 app.use(bodyParser.json());
 
 app.post('/testErrorHandling', function(req, res, next) {
   if (has(req.body, 'test') && req.body.test !== true) {
-    return next(new Error('Error on Express Regular Error POST Route'));
+    return next!(new Error('Error on Express Regular Error POST Route'));
```
Comment (ofrobots) [inline test/test-servers/express_scaffold_server.ts]: Seems unrelated. Why is this being changed as part of this PR?

File: test/test-servers/hapi_scaffold_server.ts [hunk 8]
```diff
@@ -51,7 +52,7 @@ server.route({
   },
 });
 
-server.register({register: errorHandler.hapi}, err => {
+server.register({register: errorHandler.hapi} as {} as hapi.PluginFunction<{}>, err => {
```
Comment (ofrobots) [inline test/test-servers/hapi_scaffold_server.ts]: Does this need to happen as part of this PR?

File: test/test-servers/koa_scaffold_server.ts (hunk 3 above)
Comment (ofrobots) [reply test/test-servers/koa_scaffold_server.ts]: Issues are better than TODO comments.

File: test/test-servers/hapi_scaffold_server.ts (hunk 8 above)
Comment (DominicKramer) [reply test/test-servers/hapi_scaffold_server.ts]: Yes.  With the change to use `imports`, the types don't exactly align.  I will create an issue to track this.

File: test/test-servers/koa_scaffold_server.ts (hunk 3 above)
Comment (DominicKramer) [reply test/test-servers/koa_scaffold_server.ts]: I'll create an issue.

File: test/unit/interfaces/hapi.ts (hunk 5 above)
Comment (DominicKramer) [reply test/unit/interfaces/hapi.ts]: Awesome.  Thanks.

File: test/test-servers/express_scaffold_server.ts (hunk 7 above)
Comment (DominicKramer) [reply test/test-servers/express_scaffold_server.ts]: Yes.  With the change to use `imports`, type checking indicates that `next` could be `null`.  I used `!` to cause the test that uses this code to fail if `next` is null.
